heron/api/src/scala/com/twitter/heron/streamlet/scala/converter/ScalaToJavaConverter.scala

```diff
@@ -0,0 +1,45 @@
+//  Copyright 2018 Twitter. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+package com.twitter.heron.streamlet.scala.converter
+
+import com.twitter.heron.streamlet.Context
+import com.twitter.heron.streamlet.scala.Sink
+
+/**
+  * This class transforms passed User defined Scala Functions, Sources, Sinks
+  * to related Java versions
+  */
+object ScalaToJavaConverter {
```

Comment:
I think we need to make this object be general purpose for all of the scala traits to use, for the Builder stuff lets get this code review moving before I work more on that because I will add more to this function for the Builder stuff

Reply:
`Builder` functions accept both `SerializableSupplier` and `Source`. You can use existing `toSerializableSupplier` for `SerializableSupplier`. For `Source`, `toJavaSource` function needs to be created as same as with existing `toJavaSink` function.

Reply:
Yes I was saying I would like this checked in so that I can add those functions that you are describing